### PR TITLE
Add default values to generated docs

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* modify template to add default values to generated flags and arguments
* change `--diag-log-path` implementation, so it's default value (tempdir path) is not visible on the output
* add unit test

### CODE REVIEW

File: src/snowflake/cli/api/commands/flags.py [hunk 1]
```diff
@@ -350,8 +350,15 @@ def _password_callback(value: str):
...
+
+def _diag_log_path_callback(path: str):
+    if path is not None:
+        return path
+    return tempfile.gettempdir()
+
+
 DiagLogPathOption: Path = typer.Option(
-    tempfile.gettempdir(),
+    None,
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/api/commands/flags.py:361]: Let's add a comment explaining why we do so

File: tests/__snapshots__/test_docs_generation_output.ambr [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    Database schema to use. Overrides the value specified for the connection.
+  
+  :samp:`--role, --rolename {TEXT}`
+    Role to use. Overrides the value specified for the connection.
+  
+  :samp:`--warehouse {TEXT}`
+    Warehouse to use. Overrides the value specified for the connection.
+  
+  :samp:`--temporary-connection, -x`
+    Uses connection defined with command line parameters, instead of one defined in config. Default: False
```
Comment (sfc-gh-turbaszek) [inline tests/__snapshots__/test_docs_generation_output.ambr:87]: Why sometimes there's ending `.` and sometimes not?

File: src/snowflake/cli/api/commands/flags.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/api/commands/flags.py]: Done!

File: tests/__snapshots__/test_docs_generation_output.ambr (hunk 2 above)
Comment (sfc-gh-pczajka) [reply tests/__snapshots__/test_docs_generation_output.ambr]: I left no-dot after default, but no strong opinion. Changed

**Review by sfc-gh-turbaszek — APPROVED** [review]

**Pull request merged**